src/treeview/guide/guideTreeViewProvider.ts

```diff
@@ -0,0 +1,34 @@
+import * as vscode from "vscode";
+
+import { GuideTreeItem } from "./guideTreeItem";
+
+import { AppContext } from "../../context/app";
+import { createGuideContent, GuideDocsMeta } from "../../schemas/guide";
+import { GUIDE_DOCS_META_DATA } from "../../variables";
+import { ChildTreeItem, PreviewTreeItem } from "../previewTreeItem";
+
+type TreeDataProvider = vscode.TreeDataProvider<ChildTreeItem>;
+
+export class GuideTreeViewProvider implements TreeDataProvider {
+  private readonly context: AppContext;
+  private readonly guideDocsMetaData: GuideDocsMeta[];
+
+  constructor(context: AppContext) {
+    this.context = context;
+    this.guideDocsMetaData = GUIDE_DOCS_META_DATA;
+  }
+
+  async getTreeItem(element: PreviewTreeItem): Promise<PreviewTreeItem> {
+    return element;
+  }
+
+  async getChildren(element?: PreviewTreeItem): Promise<ChildTreeItem[]> {
+    if (element) return element.getChildren();
+
+    const treeItems = this.guideDocsMetaData.map(
+      (result) => new GuideTreeItem(this.context, createGuideContent(result))
+    );
```

Comment:
上記のレビューを受け入れるなら、`GUIDE_DOCS_META_DATA` を使うようにしていただいて、`result` だと意味が伝わりづらいと思ったので、`metadata` ではどうでしょうか？

```suggestion
    const treeItems = GUIDE_DOCS_META_DATA.map(
      (metadata) => new GuideTreeItem(this.context, createGuideContent(metadata))
    );
```